src/codegen.c

```diff
@@ -0,0 +1,385 @@
+/*
```

Comment:
Can you move the handwritten part of `src/codegen.c` into `src/compile.c`? Therefore, `src/codegen.c` would be completely generated by tools, implying that we can ignore it in git tree.